bin/latexmlc

```diff
@@ -133,56 +132,43 @@ if ((!$sock) && $local && ($expire == -1)) {
     my $response = decode_json($http_response->content);
     ($result, $status, $log) = map { $$response{$_} } qw(result status log) if defined $response;
   } else {
-    print STDERR "Fatal:HTTP:" . $http_response->code() . " " . $http_response->message() . "\n"; exit 1;
+    die("Fatal:HTTP:" . $http_response->code() . " " . $http_response->message() . "\n");
   }
 }
 
 #***************************************************************************
 ### Common treatment of output:
+my $destination = $opts->get('destination');
 
-# Special features for latexmls:
+# Always write log to a file, either specified via --log or 'latexml.log'
+# with the exception of *archive* outputs, where the log is in the ZIP
 my $whatsout   = $opts->get('whatsout');
 my $is_archive = $whatsout && ($whatsout =~ /^archive/);
-if ($log) {
-  if ($opts->get('log')) {
-    if (!$is_archive) {    # archives only have the archive file
-      my $clog = $opts->get('log');
-      my $log_handle;
-      if (!open($log_handle, ">", $clog)) {
-        print STDERR "Fatal:I/O:forbidden Couldn't open log file $clog : $!\n";
-        exit 1;
-      }
-      print $log_handle $log;
-      close $log_handle;
-    }
-  } else { print STDERR "\n", $log, "\n"; }    #STDERR log otherwise
-}
+if (!$is_archive && $log) {
+  my $log_path = $opts->get('log') || 'latexml.log';
+  open(my $log_handle, ">", $log_path) or die("Fatal:I/O:forbidden Couldn't open log file $log_path : $!\n");
+  print $log_handle $log;
+  close $log_handle; }
 
```

Comment:
So, not using `UseLog()`? And always "latexml.log", rather than "$jobname.latexml.log"?  Is this part of what you're wanting to defer till later, or what you're trying to push as the "right" way?

Reply:
1. I was just learning of UseLog(), so it slipped my mind to try to use it when flushing the full `$$response{log}` from a conversion. It would look something like:
    ```perl
    UseLog($log_path);
    print $LaTeXML::Common::LOG $log;
    UseLog(undef);
    ```
which is mildly odd... Maybe we need a DumpToLog method in Error.pm that does this?

2. The `latexml.log` considerations are a bit unclear to me. Until now latexmlc didn't use a default `latexml.log` at all, so adding it in as-is is in a way a simple upgrade. But the big challenge is that in my uses of latexmlc I don't have a jobname way too often. I test around with various `literal:` formulas and fragments, for which cases I think the default `latexml.log` would be OK, or comparably a `literal.latexml.log` ?

Getting this discussion is exactly what I was hoping for actually :>

Reply:
(1) Yeah, odd. But what's odd (to me) is that you're returning the log as a string at all (in general).  If the log should go in a file, get the name, and somebody call's `UseLog($logpath)`, and you're done. In a server context where you don't want the file, use a string ref for `$logpath`, and then send the string wherever it goes.

(2) Sure, `latexml.log` is quite reasonable for the cases where you *don't* have a reasonable jobname; but when you do, not so much.

Reply:
(1) Well, an API call should return the payload, so a `$$response{log}` is completely natural from the LaTeXML.pm perspective. latexmlc may decide to write the value down to a file, ltxmojo/latexmls may decide to ship it back over the wire.

But it doesn't need to *always* get returned, and in some cases it comes back empty already -- whenever there is an archive being written with `--log` set, LaTeXML.pm would create that file and return the raw bytes for the archive in `{result}`, without setting `{log}` at all. I keep thinking of this as some sort of "interactive" flag, though still a bit fuzzy what's the least confusing way to make it configurable.

(2) Ok, I'll try to rework that part.

Reply:
Thinking on your suggestion of:
>  In a server context where you don't want the file, use a string ref for $logpath, and then send the string wherever it goes.

That does make sense after playing with the thought for a while. Ideally we at least deal with "filehandles and string refs", which look the same to Perl, rather than "pathnames and string refs" which are very different kinds of objects. Maybe we need a `UseLogHandle()` that I can call with the string ref from LaTeXML.pm, and which `UseLog()` can call after it opens the file and gets a handle on it?

Then... latexmlc can figure out a `--log` option for Config before the conversion starts, and `->convert()` can be interactive, as are latexml and latexmlpost. 

Meanwhile, latexmls/ltxmojo won't set a `--log` for most jobs (archives being the one exception), which LaTeXML.pm will interpret -- as it already does -- as a fallback to a local string ref, to be returned serialized in the response.

Yeah that can work, let me try...

Reply:
Sure, but the "payload" can be whatever you (or I) say it is: a pathname, eg, not necessarily the whole log in a string.

Currently, `UseLog` can already distinguish between a string (pathname) and string ref; there can be more cases if we can distinguish them and they're useful. 

For latexml/latexmlpost, if no `--log` was specified, I'd think the log should default to jobname based.  But I can see the ltxmojo cases want to be able to say no-log; either with an explicit `--log=none` or otherwise detect these cases.
(the word "interactive" seems to go too far)